autoload/db.vim

```diff
@@ -519,3 +602,129 @@ function! db#clobber_dbext(...) abort
   endfor
   return opts
 endfunction
+
+function! s:vim_job.cb(job, data) dict abort
+  if type(a:data) ==? type(0)
+    let self.exit = 1
+    let self.exit_status = a:data
+    return self.call_cb_if_finished()
+  endif
+  let self.output .= a:data
+endfunction
```

Comment:
Why are exit and output handled by the same callback? There's no shared code, just a conditional that teases it back into 2 separate pieces of logic.

Reply:
That's orthogonal. The code inside the if statement _only_ fires for `exit_cb`, and the code outside doesn't. `exit_cb` can/should be a separate function.

Reply:
Makes sense. I separated it to it's own function.